src/components/BottomSheetModal.tsx

```diff
@@ -0,0 +1,115 @@
+import { Trans } from '@lingui/macro'
+import { globalFontStyles } from 'css/font'
+import { forwardRef, PropsWithChildren, useState } from 'react'
+import { createPortal } from 'react-dom'
+import styled, { keyframes } from 'styled-components/macro'
+
+import Dialog, { Animation, Header, Modal, Provider as DialogProvider } from './Dialog'
+
+const slideInBottom = keyframes`
+  from {
+    transform: translateY(calc(100vh));
+  }
+`
+
+const slideOutBottom = keyframes`
+  to {
+    transform: translateY(100%);
+  }
+`
+
+const BottomSheetModalBackdrop = styled.div<{ className?: string }>`
+  background-color: ${({ theme }) => theme.scrim};
+  bottom: 0;
+  left: 0;
+  opacity: 1;
+  position: fixed;
+  right: 0;
+
+  &.hidden {
+    opacity: 0;
+    transition: visibility 0s linear 0.25s, opacity 0.25s;
+    visibility: hidden;
+  }
+
+  top: 0;
+  transition: visibility 0s linear 0s, opacity 0.25s;
+  visibility: visible;
+`
+
+const Wrapper = styled.div<{ open: boolean }>`
+  ${globalFontStyles};
```

Comment:
Move this to the Dialog's wrapping element, so that it covers all cases of portaled dialogs.